windows/vNext/ReactNativeVideoCPP/ReactVideoView.cpp

```diff
@@ -0,0 +1,255 @@
+#include "pch.h"
+#include "ReactVideoView.h"
+#include "ReactVideoView.g.cpp"
+#include "NativeModules.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Core;
+using namespace Windows::Media::Core;
+using namespace Windows::Media::Playback;
+
+namespace winrt::ReactNativeVideoCPP::implementation {
+ReactVideoView::ReactVideoView(winrt::Microsoft::ReactNative::IReactContext const &reactContext)
+    : m_reactContext(reactContext) {
+  // always create and set the player here instead of depending on auto-create logic
+  // in the MediaPlayerElement (only when auto play is on or URI is set)
+  m_player = winrt::Windows::Media::Playback::MediaPlayer();
+  SetMediaPlayer(m_player);
+  m_uiDispatcher = CoreWindow::GetForCurrentThread().Dispatcher();
+  auto mediaOpenedToken = m_player.MediaOpened([ref = get_weak()](auto const &sender, auto const &args) {
+    auto self = ref.get();
+    if (self) {
+      self->OnMediaOpened(sender, args);
+    }
+  });
+
+  auto mediaFailedToken = m_player.MediaFailed([ref = get_weak()](auto const &sender, auto const &args) {

```

Comment:
Do these events get unregistered properly? Can the auto_revoke pattern be used instead?

Reply:
It seems working fine, but I am going to apply the auto-revoke pattern just to be safe.